Move InstrumentationLibrary key generation logic out of TracerProvider/SpanConverter [#619]

ditsuke opened the pull request:
PR for Issue #619.

Factors key generation logic for InstrumentationLibrary instances out to a helper method in a utils class, replacing generation in `TracerProvider` and `SpanConverter` with calls to the new method.

ditsuke commented:
Could use some feedback on the new `Helpers` class, can we do a better name?

brettmc commented:
@ditsuke welcome aboard! re: class naming, I think `SDK\Common\Instrumentation` is a better location. As to class name, `KeyGenerator`? It will also need unit tests.

ditsuke commented:
> @ditsuke welcome aboard! re: class naming, I think `SDK\Common\Instrumentation` is a better location. As to class name, `KeyGenerator`? It will also need unit tests.

@brettmc thanks! I've moved and renamed the generator and added unit tests, PTAL

tidal commented on tests/Unit/SDK/Common/Instrumentation/KeyGeneratorTest.php:
```diff
@@ -0,0 +1,34 @@
+        $key = KeyGenerator::generateInstanceKey($name, null, null);
+
+        $this->assertSame(sprintf('%s@unknown ', $name), $key);
+    }
```
Could you please also add a test for providing a `name`, `version` and no `schemaURL`?
This is only to assert, that key are different for all of these three variants.

tidal commented:
@ditsuke 
Thank you. LGTM
I have added one little comment for a test you could add, just to make sure.
However other than his, I think we can merge the PR, if you are ok with it.

ditsuke commented:
> @ditsuke Thank you. LGTM I have added one little comment for a test you could add, just to make sure. However other than t, I think we can merge the PR, if you are ok with it.

@tidal thanks for reviewing. I've added the third test with b84e3bb

ditsuke commented:
Thanks!
